io/zenoh-transport/src/unicast/link.rs

```diff
@@ -177,13 +193,33 @@ async fn tx_task(
     link: LinkUnicast,
     keep_alive: Duration,
     #[cfg(feature = "stats")] stats: Arc<TransportUnicastStatsAtomic>,
+    #[cfg(feature = "transport_compression")] compression_enabled: bool,
 ) -> ZResult<()> {
+    #[cfg(feature = "transport_compression")]
+    let mut compression_aux_buff: Box<[u8]> = vec![0; usize::pow(2, 16)].into_boxed_slice();
+
     loop {
         match pipeline.pull().timeout(keep_alive).await {
             Ok(res) => match res {
                 Some((batch, priority)) => {
                     // Send the buffer on the link
-                    let bytes = batch.as_bytes();
+                    #[allow(unused_mut)]
+                    let mut bytes = batch.as_bytes();
+
+                    #[cfg(feature = "transport_compression")]
+                    let compression: Vec<u8>;
```

Comment:
I believe this is a non-valid Rust statement since the vector is non-initialized.

Reply:
Good to know, it's the first time I see it indeed :)